stor/dx.py

```diff
@@ -113,96 +165,727 @@ def temp_url(self, lifetime=300, filename=None):
+
+        The source cannot be the root directory.
+
+        Refer to copytree or copytree for detailed information.
+
+        Args:
+            dest (Path): The destination directory path within same project
+
+        Raises:
+            TargetExistsError: When destination directory already exists
+            DNAnexusError: When attempting to move across projects
+        """
+        if dest.canonical_project != self.canonical_project:
+            raise DNAnexusError('Cannot movetree across different projects')
+        if not self.resource:
+            raise DNAnexusError('Cannot move root folder within same project on DX')
+        if self == dest:
+            return
+        if dest == (self.drive+dest.project):  # need to convert dx://proj to dx://proj:
+            dest = dest + ':'
+        target_dest, should_rename, moved_folder_path = self._prep_for_copytree(dest)
+
+        project_handler = dxpy.DXProject(self.canonical_project)
+        with _propagate_dx_exceptions():
+            project_handler.move_folder(
+                folder=self.resource,
+                destination=target_dest.parent.resource or '/'
+            )
+            if should_rename:
+                dxpy.api.project_rename_folder(
+                    dest.canonical_project,
+                    input_params={
+                        'folder': moved_folder_path.resource,
+                        'name': target_dest.name
+                    }
+                )
+        self.clear_cached_properties()
+
+    @_propagate_dx_exceptions()
+    def download_object(self, dest, **kwargs):
+        """Download a single path or object to file.
+
+        Args:
+            dest (Path): The output file
+
+        Raises:
+            NotFoundError: When source path is not an existing file
+        """
+        dxpy.download_dxfile(
+            dxid=self.canonical_resource,
+            filename=dest,
+            project=self.canonical_project
+        )
+
+    def download_objects(self,
+                         dest,
+                         objects):
+        """Downloads a list of objects to a destination folder.
+
+        Note that this method takes a list of complete relative or absolute
+        paths to objects (in contrast to taking a prefix). If any object
+        does not exist, the call will fail with partially downloaded objects
+        residing in the destination path.
+
+        Args:
+            dest (str): The destination folder to download to. The directory
+                will be created if it doesnt exist.
+            objects (List[str|PosixPath|SwiftPath]): The list of objects to
+                download. The objects can paths relative to the download path
+                or absolute dx paths. Any absolute dx path must be
+                children of the download path
+
+        Returns:
+            dict: A mapping of all requested ``objs`` to their location on
+                disk
+
+        Examples:
```

Comment:
This is exactly the same as `swift.SwiftPath.download_objects` right? Maybe better to move to central location rathe rthan duplicate method information

Reply:
Fixed.